src/test/java/io/cdap/plugin/switchcase/route/RoutingSwitchTest.java

```diff
@@ -0,0 +1,150 @@
+ * the License.
+ */
+
+package io.cdap.plugin.switchcase.route;
+
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.InvalidEntry;
+import io.cdap.cdap.etl.api.SplitterTransform;
+import io.cdap.cdap.etl.mock.common.MockMultiOutputEmitter;
+import io.cdap.cdap.etl.mock.transform.MockTransformContext;
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Tests for {@link RoutingSwitch} that apply to all {@link PortSpecification port specifications}
+ */
+public abstract class RoutingSwitchTest {
+  static final Schema INPUT =
+    Schema.recordOf("input",
+                    Schema.Field.of("supplier_id", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                    Schema.Field.of("part_id", Schema.of(Schema.Type.STRING)),
+                    Schema.Field.of("count", Schema.of(Schema.Type.INT)));
+  static final String PORT_SPECIFICATION = "a_port:equals(10)";
+
+  // TODO: Add an abstract method getMode(), so test can be executed for all modes after adding jexl support
+
+  @Test
+  public void testDefaultedRecordToErrorByDefault() throws Exception {
+    testDefaultedRecord(null, null);
+  }
+
+  @Test
+  public void testDefaultedRecordToError() throws Exception {
+    testDefaultedRecord(RoutingSwitch.Config.DefaultHandling.ERROR_PORT.value(), null);
+  }
+
+  @Test
+  public void testDefaultedRecordToDefaultDefaultPort() throws Exception {
+    testDefaultedRecordToDefaultPort(null);
+  }
+
+  @Test
+  public void testDefaultedRecordToDefaultPort() throws Exception {
+    testDefaultedRecordToDefaultPort("Custom Default Port");
+  }
+
+  @Test
+  public void testDefaultedRecordSkipped() throws Exception {
+    testDefaultedRecord(RoutingSwitch.Config.DefaultHandling.SKIP.value(), null);
+  }
+
+  @Test
+  public void testNullRecordSkipped() throws Exception {
+    testDefaultedRecord(RoutingSwitch.Config.NullHandling.SKIP.value(), null);
+  }
+
+  @Test
+  public void testRouting() throws Exception {
+    StructuredRecord testRecord1 = StructuredRecord.builder(INPUT)
+      .set("supplier_id", "supplier1")
+      .set("part_id", "1")
+      .set("count", "10")
+      .build();
+
+    StructuredRecord testRecord2 = StructuredRecord.builder(INPUT)
+      .set("supplier_id", "supplier2")
+      .set("part_id", "2")
+      .set("count", "20")
+      .build();
+
+    StructuredRecord testRecord3 = StructuredRecord.builder(INPUT)
+      .set("supplier_id", "supplier3")
+      .set("part_id", "3")
+      .set("count", "30")
+      .build();
+
+    String portSpecification = "Supplier 1:equals(supplier1),Supplier 2:equals(supplier2),Supplier 3:equals(supplier3)";
+    RoutingSwitch.Config config = new RoutingSwitch.Config(
+      "supplier_id", portSpecification, RoutingSwitch.Config.DefaultHandling.DEFAULT_PORT.value(), "Default Port",
```

Comment:
we should have test for all the types as routing field.

Reply:
yeah, I think we need to add some functions for the types first. I think what I'll do is remove all the types from this PR other than String, and add all the other types in a separate PR, along with some functions for them.

Reply:
I think its fine. If its going to come right after this PR.